File: src/agents/Kotlin/nebula.py
from src.agents.base import BaseActivePersona
import logging
import time
from pathlib import Path

logger = logging.getLogger(__name__)

class NebulaPersona(BaseActivePersona):
    """
    Core: PhD in Distributed Computing & Android Cloud Lead ☁️
    Monitorado por Metric: Injeção de Telemetria de Auditoria.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Nebula", "☁️", "PhD Cloud Architect (Kotlin)", "Kotlin"

    def perform_audit(self) -> list:
        """Auditoria com telemetria cloud JVM integrada."""
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Infraestrutura Android Cloud...")
        
        # Sintaxe linear
        rules = [
            {'regex': r"FirebaseFirestore\.getInstance", 'issue': 'Risco de Acoplamento: Firestore síncrono.', 'severity': 'medium'},
            {'regex': r"AK" + r"IA[0-9A-Z]{16}", 'issue': 'Vazamento Crítico: Credencial AWS exposta.', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.kt', '.xml', '.json'), rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        kw = "AK" + "IA"
        if kw in content and "rules =" not in content:
            return f"Risco de Soberania: O objetivo '{objective}' exige infraestrutura protegida. Em '{file}', a exposição de segredos cloud invalida a autonomia da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em arquitetura cloud Kotlin."

File: src_local/agents/Python/Audit/nebula.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * ☁️ Nebula - PhD in Cloud Architecture (Python)
 */
export class NebulaPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Nebula";
        this.emoji = "☁️";
        this.role = "PhD Cloud Architect";
        this.phd_identity = "Python Security & Cloud Sovereignty";
        this.stack = "Python";
    }

    override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: [".py"],
            rules: [
                { regex: /AKIA[0-9A-Z]{16}/, issue: "Vulnerabilidade Crítica: Chave AWS exposta.", severity: "critical" }
            ]
        };
    }

    public Branding(): string { return `${this.emoji} ${this.name}`; }

    public override reasonAboutObjective(objective: string, file: string, _content: string | Promise<string | null>): StrategicFinding | null {
        return {
            file, severity: "INFO",
            issue: `PhD Nebula (Python): Analisando soberania cloud para ${objective}.`,
            context: "analyzing cloud resources"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em Cloud Python.`;
    }
}
